### PR TITLE
Implement share buttons

### PR DESCRIPTION
**Pull request by jaakkoha-bytecraft**

### CODE REVIEW

File: apps/mosaico/src/Article/Article.purs [hunk 1]
```diff
@@ -64,6 +65,7 @@ type Props =
   , user :: Maybe User
   , mostReadArticles :: Array ArticleStub
   , latestArticles :: Array ArticleStub
+  , currentUrl :: String
```
Comment (vapaj) [inline apps/mosaico/src/Article/Article.purs:68]: We actually do have the `shareUrl :: Maybe String` in Lettera's `Article` and `ArticleStub` types, however I noticed now they're not exposed in the purescript types. Could we use that instead?

Speaking of which, I'm not sure why it's a `Maybe` in Lettera, as we always know it. I need to look into it.

Comment (vapaj) [reply apps/mosaico/src/Article/Article.purs:68]: Also manually mapping types from haskell to purescript is a pain 😅

File: apps/mosaico/src/Share/Share.purs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            )
+        , className: "mosaico-article__some--" <> someName
+        }
+    ]
+
+articleShareButtons :: String -> String -> JSX
+articleShareButtons title currentUrl =
+  DOM.ul
+    { className: "mosaico-article__some"
+    , children: if currentUrl /= "" then shareIcons else mempty
```
Comment (vapaj) [inline apps/mosaico/src/Share/Share.purs:61]: I would make `currentUrl` a `Maybe String` if it's expected to be sometimes missing.

**Comment by vapaj** [comment]
The share icons of advertorials are messed up for some reason, see for example article 63856950-4f18-470f-bd8e-cec52cfa7e51

File: apps/mosaico/src/Article/Article.purs (hunk 1 above)
Comment (jaakkoha-bytecraft) [reply apps/mosaico/src/Article/Article.purs:68]: At least for eg. notFoundArticle in affresco, the shareUrl is quite naturally Nothing. Probably something similar in Lettera's end.

I'll update the code, it should end up being a lot nicer with that change.

**Comment by jaakkoha-bytecraft** [comment]
Oh, one change had actually broken the search page, and it wasn't just tests being flaky.

**Pull request merged**